be/ut/alloc.c

```diff
@@ -398,6 +398,204 @@ M0_INTERNAL void m0_be_ut_alloc_spare(void)
 	M0_LEAVE();
 }
 
+M0_INTERNAL void m0_be_ut_alloc_align(void)
+{
+	/**
+	 * In this UT, we are testing the functionality of m0_be_alloc_aligned()
+	 * when chunk_align parameter is set to true.
+	 *
+	 * 1. Allocate multiple chunks with chunk_align set to true.
+	 * 2. Verify the alignment of all the allocated chunks.
+	 * 3. Delete some of the allocated chunks.
+	 * 4. Verify the alignment of the remaining chunks.
+	 * 5. Delete remaining of the allocated chunks.
+	 */
+	struct m0_be_ut_backend *ut_be    = &be_ut_alloc_backend;
+	struct m0_be_ut_seg     *ut_seg   = &be_ut_alloc_seg;
+	struct m0_be_allocator  *a;
+	void                    *ut_ptr[BE_UT_ALLOC_PTR_NR];
+	int                      ut_nr    = BE_UT_ALLOC_PTR_NR;
+	uint64_t                 j;
+	int                      i;
+	int                      ut_shift = BE_UT_ALLOC_SHIFT - 1;
+	int                      ut_size;
+	bool                     ut_tval;
+
+	m0_be_ut_backend_init(ut_be);
+	m0_be_ut_seg_init(ut_seg, ut_be, BE_UT_ALLOC_SEG_SIZE);
+	m0_be_ut_seg_allocator_init(ut_seg, ut_be);
+
+	a = m0_be_seg_allocator(ut_seg->bus_seg);
+	M0_UT_ASSERT(a != NULL);
+	M0_SET_ARR0(ut_ptr);
+
+	/* Alloc chunks with chunk_align parameter set as true */
+	for (i = 0; i < ut_nr; i++) {
+		j = i;
+		ut_size = m0_rnd64(&j) % BE_UT_ALLOC_SIZE + 1;
+
+		M0_BE_UT_TRANSACT(ut_be, tx, cred,
+			  (m0_be_allocator_credit(a, M0_BAO_ALLOC_ALIGNED,
+						  ut_size, ut_shift, &cred),
+			   m0_be_alloc_stats_credit(a, &cred)),
+			  (M0_BE_OP_SYNC(op,
+				 m0_be_alloc_aligned(a, tx, &op, &ut_ptr[i],
+						     ut_size, ut_shift,
+						     M0_BITS(M0_BAP_NORMAL),
+						     true)),
+			   m0_be_alloc_stats_capture(a, tx)));
+		M0_UT_ASSERT(ut_ptr[i] != NULL);
+	}
+
+	/* Verify the alignment of the chunks */
+	for (i = 0; i < ut_nr; i++)
+		M0_UT_ASSERT(m0_addr_is_aligned(ut_ptr[i], ut_shift));
+
+	/* Delete the even numbered chunks */
+	for (i = 0; i < ut_nr; i += 2) {
+		j = i;
+		ut_size = m0_rnd64(&j) % BE_UT_ALLOC_SIZE + 1;
+
+		M0_BE_UT_TRANSACT(ut_be, tx, cred,
+			  (m0_be_allocator_credit(a, M0_BAO_FREE_ALIGNED,
+						  ut_size, ut_shift, &cred),
+			   m0_be_alloc_stats_credit(a, &cred)),
+			  (M0_BE_OP_SYNC(op,
+					 m0_be_free_aligned(a, tx, &op,
+							    ut_ptr[i])),
+			   m0_be_alloc_stats_capture(a, tx)));
+		ut_ptr[i] = NULL;
+	}
+
+	/* Verify the alignment of the remaining chunks */
+	for (i = 0; i < ut_nr; i++) {
```

Comment:
If you have deleted some allocations then you need to cut the iteration of this loop.